syntax = "proto3";

package raft.external;

message PutRequest {
  string key = 1;
  string value = 2;
  string client = 3;
}

message PutReply {
  bool status = 1;
}

message GetRequest {
  string key = 1;
  string client = 2;
}

message GetReply {
  bool status = 1;
  string value = 2;
}

message Reply {
}

service External {
  rpc Put (PutRequest) returns (Reply) {}

  rpc Get (GetRequest) returns (Reply) {}

  rpc ReplyPut (PutReply) returns (Reply) {}

  rpc ReplyGet(GetReply) returns (Reply) {}
}
